api/src/main/java/org/pmiops/workbench/api/OfflineRuntimeController.java

```diff
@@ -357,14 +352,20 @@ private boolean notifyForUnusedDisk(LeonardoListPersistentDiskResponse disk, int
     return true;
   }
 
-  private boolean isDiskAttached(
-      LeonardoListPersistentDiskResponse diskResponse, String googleProject) throws ApiException {
+  private boolean isDiskAttached(ListPersistentDiskResponse diskResponse, String googleProject)
+      throws ApiException {
     final String diskName = diskResponse.getName();
 
     if (leonardoMapper.toApiListDisksResponse(diskResponse).getIsGceRuntime()) {
       return runtimesApiProvider.get().listRuntimesByProject(googleProject, null, false).stream()
-          .flatMap(runtime -> Stream.ofNullable(runtime.getDiskConfig()))
-          .anyMatch(diskConfig -> diskName.equals(diskConfig.getName()));
+          .flatMap(
```

Comment:
is this equivalent?  maybe it is, but it's not obvious - let's take a look.

Reply:
so I think the previous code has a bug...cuz getRuntime doesn't actually return `diskName` today. This is what the actual payload from leo looks like
```
    "runtimeConfig": {
      "machineType": "n1-standard-4",
      "persistentDiskId": 13259,
      "cloudService": "GCE",
      "bootDiskSize": 120,
      "zone": "us-central1-b",
      "gpuConfig": null,
      "configType": "GceWithPdConfig"
    }
```

So the updated code should be correct cuz it's comparing the diskId against the id that's returned from `listDisk`